types/docs/README.md

```diff
@@ -0,0 +1,115 @@
+# Documentation
+
+## DomainType
+
+`DomainType` is a sequence of 4 bytes that uniquely identifies the network:
+- The first two bytes are yet empty and lack meaning.
+- The third byte is the network ID (e.g. Mainnet, Primus, Shifu, Jato)
+- The fourth byte is the current fork number of the network.
+
+## BeaconNetwork
+
+`BeaconNetwork` is a type that allows accessing information about the blockchain, e.g. such as the current slot, the number of seconds per slot, and the fork version.
```

Comment:
Based on what I see in the code,

perhaps worth noting that `BeaconNetwork` does not need to send requests to ETH2 node (or any requests at all) and rather just computes this data based on a bunch of constants (and values provided to it when asked to do this computation).